packages/ui/src/components/Popup/index.tsx

```diff
@@ -422,6 +422,49 @@ export const Popup = forwardRef(
       hideOnClickOutside,
     ])
 
+    /**
+     * This event will occur only for dialog and will trap focus inside the dialog.
+     */
+    const handleFocusTrap: KeyboardEventHandler = useCallback(event => {
+      event.stopPropagation()
+      const isTabPressed = event.key === 'Tab'
+
+      if (!isTabPressed) {
+        return
```

Comment:
Maybe stop propagation after testing if tab is pressed, so the event can properly bubble?

Reply:
@ModuloM good point I took this from Modal made by dorian without re-thinking the code, I will change it